File: spring-cloud-integrated-demo/membership-service/src/test/java/com/github/jccode/springcloud/integrateddemo/membership/service/MemberPointFlowServiceTest.java
package com.github.jccode.springcloud.integrateddemo.membership.service;

import com.github.jccode.springcloud.integrateddemo.membership.model.MemberPointFlow;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mybatis.spring.boot.test.autoconfigure.MybatisTest;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.annotation.Import;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit4.SpringRunner;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.greaterThan;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.Matchers.nullValue;

@SpringBootTest
@RunWith(SpringRunner.class)
@Import(MemberPointFlowService.class)
@MybatisTest
@ActiveProfiles("test")
public class MemberPointFlowServiceTest {

    @Autowired
    private MemberPointFlowService memberPointFlowService;

    @Test
    public void mybatisWorks() {
        MemberPointFlow flow = new MemberPointFlow();
        flow.setUserId(1);
        flow.setOrderId(1);
        flow.setPoint(10);
        assertThat(flow.getId(), is(nullValue()));

        int ret = memberPointFlowService.save(flow);
        assertThat(ret, greaterThan(0));
        assertThat(flow.getId(), greaterThan(0));
    }
}

File: spring-cloud-integrated-demo/membership-service/src/test/scala/com/github/jccode/springcloud/integrateddemo/membership/service/MemberPointFlowServiceTest.scala
package com.github.jccode.springcloud.integrateddemo.membership.service


import com.github.jccode.springcloud.integrateddemo.membership.model.MemberPointFlow
import org.junit.Test
import org.junit.runner.RunWith
import org.mybatis.spring.boot.test.autoconfigure.MybatisTest
import org.scalatest.Matchers
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.context.annotation.Import
import org.springframework.test.context.ActiveProfiles
import org.springframework.test.context.junit4.SpringRunner

@SpringBootTest
@RunWith(classOf[SpringRunner])
@Import(Array(classOf[MemberPointFlowService]))
@MybatisTest
@ActiveProfiles(Array("test"))
class MemberPointFlowServiceTest extends Matchers {

  @Autowired
  private val memberPointFlowService: MemberPointFlowService = null

  @Test
  def mybatisWorks(): Unit = {
    val flow = new MemberPointFlow
    flow.setUserId(1)
    flow.setOrderId(1)
    flow.setPoint(10)
    flow.getId should be (null)
    val ret = memberPointFlowService.save(flow)
    ret should be >0
    flow.getId.toInt should be >0
  }
}
